app/models/activity.rb

```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+class Activity < ApplicationRecord
+  extend FriendlyId
+  include Ownable
+  include Finishable
+  include SearchActivities
+
+  belongs_to :garden, optional: true
+  belongs_to :planting, optional: true
+
+  friendly_id :activity_slug, use: %i(slugged finders)
+
+  CATEGORIES = ["General", "Weeding", "Soil Cultivation", "Fertilizing", "Pruning", "Watering"]
```

Comment:
Suggestions welcome for any extras here

Reply:
Is it worth distinguishing types of fertilising? E.g. giving the tomatoes liquid food or giving them more compost are both technically fertilising. Would the latter be soil cultivation? The rest sounds good.

Reply:
We've already tracked harvesting separately, but should harvesting be changed to fit in this activity model?

Reply:
Brenda also mentioned in #1165 about treating diseases - what are the sensible activities for that? I think the categories we have here cover everything needed on #3451.

Reply:
> harvesting be changed to fit in this activity model?

Maybe long term it should have an interface to this. Sort of does via timeline service.

Or these activities should have their own models whenever their attributes are unique. IE a watering event has one duration/volume/source/salinity/etc.
A harvest has one or more crops, volume

Reply:
For diseases, let's add a generic 'treatment'.

That covers topical application of many, many things, as well as more general "I applied a growth aid kind of like fertilizer, but not technically one"

Reply:
Liquid vs compost vs solid tablet vs blood and bone, I would put "mix compost into soil" or "mulching" as general soil cultivation activities, with the others specifically fertilizing.